### PR TITLE
pacifist antags retain pacifism

### PR DESCRIPTION
**Pull request by imsxz**
## About The Pull Request

Just deletes a few lines of code.

## Why It's Good For The Game

Pacifism is an opt-in negative quirk. It can be worked around in many ways on the off-chance you get nukie or something as well. It's essentially free trait points when you're antag for whatever it's worth.

## Changelog
:cl: imsxz
tweak: antags with the pacifist trait no longer lose pacifism.
/:cl:

### CODE REVIEW

**Comment by subject217** [comment]
merge conflict 4head

**Comment by imsxz** [comment]
@tgstation-server fucking tell me i have a merge conflict 1 more god damn time you cheeky ass 2 dollar whore i swear to god

**Comment by wesoda25** [comment]
Friendly antags are a cancer and should not be encouraged. 

At LEAST make it so that pacifism is deleted upon becoming wizard. They're the cancer cell at the top of the throne.

**Comment by imsxz** [comment]
you can still murderbone as a pacifist dummy it just takes a bit more brainpower

**Comment by subject217** [comment]
it doesn't really encourage or discourage friendly antags fam, people will do what they want to do, trait or no trait

**Comment by tralezab** [comment]
Don't give objectives that kill if the traitor has pacifism

**Comment by tralezab** [comment]
As I see it, some antagonists don't make sense/work as pacifists and the ones that do haven't been edited so that they work with pacifism. so probably not a great idea

**Comment by PKPenguin321** [comment]
having to exploit the jankily-coded pacifism trait to kill people is stupid in the first place and should not be encouraged, it goes against the spirit of the feature (in that you're supposed to be a pacifist). the intended use of the pacifist trait conflicts with pretty much all antags and should remain separate.

**Comment by wesoda25** [comment]
imsxz you’re literally one of the most robust players on the server. What you say can’t apply to everyone else because everyone else hasn’t reached the point where they need to impose challenges upon themselves. If anything this’ll just make it so that people will code even more restrictions in for pacifism, since you clearly mean to show how flawed it is.

**Comment by tralezab** [comment]
Make it a setting to keep pacifism as an antagonist and give non assassination objectives (turn them into prevent escape!) OR they can disable it when they become a pacifist and get full objectives

**Comment by nicbn** [comment]
Pacifism as non antag is opt-in and so is pacifism as antag: just take some pax, add an admin or simply don't attack anyone. But opting for non-antag pacifism shouldn't obligate you into antag pacifism

**Comment by dr-bee** [comment]
When you pick traits you really shouldnt be separating "antag" from "non-antag". You are picking character traits, not "non-antag" traits. If you dont want to risk having to roleplay around pacifism as an antag the dont pick the trait. Exempting antags from negative aspects of the trait system is un-intuative as hell and is a buff they dont really neeed.

**Comment by imsxz** [comment]
you could make the same argument for all negative quirks. If people want a debuff, they shouldn't be babied when they get antag, they should deal with it and work around it. Antags can have blindness ffs which makes it much harder than pacifism IMO.

**Comment by Tlaltecuhtli** [comment]
having steal objectives only seems cool, nothing prevents you from being a cripple blind op so why wouldnt pacifism be kept? you could like steal the disk and run away if you are the last op for minor victory because the nuke should have a pacifist check

**Comment by wesoda25** [comment]
Ok but blacklist wizard from traits.

**Comment by ExcessiveUseOfCobblestone** [comment]
I wouldn't mind a toggle between all selected/none traits while antag but imo you shouldn't be able to opt in to bad traits that get nulled while antag while also keeping the positive traits.

If they're important enough to have a balancing mechanic around them then that mechanic should be retained.

**Comment by ATH1909** [comment]
> I stand by what I said in the thread; we should have a separate traits list for round-start antag. That way people can do their RP thing when they're just part of the crew, and not possibility gimp themselves inadvertently when they're traitor, and not get the free extra positive traits as currently happens.
> 
> For people who become antag mid-round (cult or whatever), I agree they get stuck with what they had before.

I think this is the best solution.

Plus, this means that we could have antag-only quirks (that could influence objective weighting, whether or not you remain clumsy at roundstart as a clown (that quirk would probably have to be a +0 point one, as I dunno what should happen if you rolled non-clown antag with it) , etc.), which'd be pretty sweet.

**Comment by tralezab** [comment]
We've given a lot of better ways to do this but it seems like we're just gonna stick to pacifist wizards that get hijack objectives ⛳🏌️‍♂️

**Comment by zxaber** [comment]
> inb4 checking what quirks someone has to determine if they're a tot

I mean, you can already sorta do this if you see someone in a job they'd never normally play, since the game will stick you basically wherever if your preferred job(s) are taken and you're a tator.

Out of scope for this PR, but the real question is why are perks even viewable on the medical scanner to start with? Is there any particular reason anyone else needs to know that I like to sleep?

**Comment by imsxz** [comment]
@tralezab out of the scope of my own coding ability, you're free to make your own PR if you think that people who willingly debuff themselves deserve to be babied as antag.

Even without this pull request I can't see why anyone would take pacifism if they find themselves "needing" to kill people lots of the time. If antags can retain blindness, legless, brain tumor,blood loss, etc, there really shouldn't be any issue with antags keeping pacifism. I'd argue it'd be much harder to kill someone as a blind antag than as a pacifist antag.

**Comment by tralezab** [comment]
I don't think "I'm not good enough to code this feature" is a valid excuse for adding a feature incorrectly, because I think some antagonists can work with pacifism but not all.

**Comment by imsxz** [comment]
this isn't a PR to add a feature fam nothing is being added let alone incorrectly. if you believe your way is better, you're more than free to

**Comment by JJRcop** [comment]
What's 4d2de5a706454b1ae429ac2a8c794a697444824f doing here?

File: code/datums/traits/negative.dm [hunk 1]
```diff
@@ -260,10 +260,40 @@
 	lose_text = "<span class='notice'>You think you can defend yourself again.</span>"
 	medical_record_text = "Patient is unusually pacifistic and cannot bring themselves to cause physical harm."
 
-/datum/quirk/nonviolent/on_process()
-	if(quirk_holder.mind && LAZYLEN(quirk_holder.mind.antag_datums))
-		to_chat(quirk_holder, "<span class='boldannounce'>Your antagonistic nature has caused you to renounce your pacifism.</span>")
-		qdel(src)
+/datum/quirk/paraplegic
```
Comment (JJRcop) [inline code/datums/traits/negative.dm]: https://github.com/tgstation/tgstation/blob/4d2de5a706454b1ae429ac2a8c794a697444824f/code/datums/traits/negative.dm#L263-L296

File: code/datums/traits/negative.dm [hunk 2]
```diff
@@ -370,3 +400,114 @@
 		if(prob(1))
 			new/obj/item/reagent_containers/food/snacks/spaghetti/pastatomato(get_turf(H)) //now that's what I call spaghetti code
 
+//If you want to make some kind of junkie variant, just extend this quirk.
+/datum/quirk/junkie
```
Comment (JJRcop) [inline code/datums/traits/negative.dm]: https://github.com/tgstation/tgstation/blob/4d2de5a706454b1ae429ac2a8c794a697444824f/code/datums/traits/negative.dm#L403-L513

**Review by JJRcop — COMMENTED** [review]
So what's all this then?

**Comment by imsxz** [comment]
changes that are already in the game i was just resolving conflicts

**Comment by ExcessiveUseOfCobblestone** [comment]
lol u broke the diff reader :(

**Comment by optimumtact** [comment]
The way I see it, if you lose your negative quirks you should lose your positive ones, and if you don't, then you should keep both.

**Comment by Jordie0608** [comment]
I don't think it's good to merge as a stop-gap.

**Comment by Tlaltecuhtli** [comment]
out of station antags shouldnt have quirks tbh

**Comment by imsxz** [comment]
i wouldn't call this a stop-gap, pacifism is the only quirk that gets removed when you get antag, this PR makes it in line with oranges decision A. I do agree that non-crew antags such as nukie abductor etc shouldn't have any quirks, though.

**Pull request merged**